components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1491,6 +1494,77 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         await this.guardAccess({ kind: "team", subject: team, members }, op);
     }
 
+    async getProviderRepositoriesForUser(params: { provider: string }): Promise<ProviderRepository[]> {
+        this.checkUser("getProviderRepositoriesForUser");
+        // Note: this operation is per-user only, hence needs no resource guard
+        return [];
+    }
+
+    public async createProject(params: CreateProjectParams): Promise<Project> {
+        this.checkAndBlockUser("createProject");
+        const span = opentracing.globalTracer().startSpan("createProject");
+        const { name, cloneUrl, teamId, appInstallationId } = params;
+        // Anyone who can read a team's information (i.e. any team member) can create a new project.
+        await this.guardTeamOperation(teamId, "get");
+        return this.projectDB.trace({ span }).storeProject(Project.create({name, cloneUrl, teamId, appInstallationId}));
+    }
+
+    public async getProjects(teamId: string): Promise<ProjectInfo[]> {
+        this.checkUser("getProjects");
+        const span = opentracing.globalTracer().startSpan("getProjects");
+        span.setTag("teamId", teamId);
+        await this.guardTeamOperation(teamId, "get");
+        return this.projectDB.trace({ span }).findProjectsByTeam(teamId);
```

Comment:
Strangely, the dashboard now fails to load projects, and shows this in the console:

```
Uncaught (in promise) Error: Request getProjects failed with message: this.projectDB.trace is not a function
```

Reply:
I'm calling this a bad deployment.

Reply:
Re-deploying didn't help 😕

Reply:
Removing the tracing for now: It doesn't work.

Also, it doesn't really make sense to trace an extremely small, new DB, while all the other DBs are not traced (except WorkspaceDB, the biggest one, where tracing makes the most sense).